### DISCUSSION TITLE
How to build app for Mac and windows with electron builder?

### DISCUSSION THREAD

**Discussion by kevinwt95** [quasarframework/quasar#15435] [CLI - Electron mode] [answered]
Hi,
I used packager before and with it everything worked without problems, but since I want to use updater in the future, I want to switch to builder.
The command `quasar build -m electron -T all` builds the app for all platforms, but I only want to build for Mac and windows.
The command `quasar build - electron -T darwin -T win32` worked with the packager before, but with the builder it only builds for mac.
Can somebody help me?

**Comment by dongwa** [comment, marked as answer]
electron-builder doesn't have the ability to cross-build, it can only build Mac apps on a Mac device and build Windows apps on a Windows device.

**Reply by TobyMosque** [reply]
You can use docker images to build to linux and windows.
https://www.electron.build/multi-platform-build.html
But since Apple has cursed us all, you'll need a mac to do a mac build.

**Comment by AlitleDingo** [comment]
You can use github actions to do this. this is what i have done sometimes.

**Reply by mkaddourcontactcare** [reply]
Can you please share how you managed this?

**Reply by AlitleDingo** [reply]
I just created a GitHub Action using a YAML file I made. The action ran successfully and outputted the DMG file.
To do this, open the repository where you want to generate the DMG file, go to the Actions tab, click Create from scratch, and paste the contents of the YAML file there.
You’ll probably want to change some parts of the YAML file, such as the name, brand, and other details.

[main.yml](https://github.com/user-attachments/files/32399340/main.yml)

**Reply by AlitleDingo** [reply]
Basically, when the YAML file runs, GitHub spins up a macOS machine, executes all the necessary npm commands, builds your project, and then uploads the final DMG file for you.
